File: C++/1518. Water Bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int drinkBottles=numBottles;
        while(numBottles>=numExchange){
            int exchangedBottles=numBottles/numExchange;
            int extraEmptyBottles=numBottles%numExchange;
            drinkBottles+=exchangedBottles;
            numBottles=exchangedBottles+extraEmptyBottles;
        }
        return drinkBottles;
    }
};

File: JAVA/1518. Water Bottles.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int sum = numBottles;
        int empty = 0;        
        while (numBottles + empty >= numExchange) {
            int temp = numBottles + empty;
            numBottles = temp / numExchange;
            empty = temp % numExchange;
            sum += numBottles;
        }        
        return sum;
    }
}
